File: project/block_manager/services/nodes/frameworks/pytorch/nodes/relu.py
"""PyTorch ReLU Activation Node Definition"""

from typing import Any, Dict, List, Optional

from ....core.types import ConfigField, Framework, NodeMetadata, TensorShape
from ..base import PyTorchPassthroughNode


class ReLUNode(PyTorchPassthroughNode):
    """Rectified Linear Unit activation"""

    @property
    def metadata(self) -> NodeMetadata:
        return NodeMetadata(
            type="relu",
            label="ReLU",
            category="basic",
            color="var(--color-accent)",
            icon="Lightning",
            description="Rectified Linear Unit activation",
            framework=Framework.PYTORCH
        )

    @property
    def config_schema(self) -> List[ConfigField]:
        return []  # No configuration needed

File: packages/core/src/lib/nodes/definitions/pytorch/relu.js
/**
 * PyTorch ReLU Activation Node Definition
 */
import { PassthroughNodeDefinition } from '../../base';
import { BackendFramework } from '../../contracts';
export class ReLUNode extends PassthroughNodeDefinition {
    constructor() {
        super(...arguments);
        Object.defineProperty(this, "metadata", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: {
                type: 'relu',
                label: 'ReLU',
                category: 'activation',
                color: 'var(--color-accent)',
                icon: 'Pulse',
                description: 'ReLU activation function',
                framework: BackendFramework.PyTorch
            }
        });
        Object.defineProperty(this, "configSchema", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: [
                {
                    name: 'inplace',
                    label: 'In-place Operation',
                    type: 'boolean',
                    default: false,
                    description: 'Perform operation in-place to save memory'
                }
            ]
        });
    }
}
